### PR TITLE
Add utils for testing Optional, bulk-export polling, and http requests.

### PR DESCRIPTION
**Pull request by michaelabuckley**
Some minor test utilities.

### CODE REVIEW

File: hapi-fhir-base/src/main/java/ca/uhn/fhir/util/MethodOutcomeUtil.java [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+import ca.uhn.fhir.rest.api.MethodOutcome;
+
+import javax.annotation.Nonnull;
+import java.util.List;
+import java.util.Optional;
+
+/**
+ * Extra methods for {@link MethodOutcome}.
+ */
+public class MethodOutcomeUtil {
```
Comment (jamesagnew) [inline hapi-fhir-base/src/main/java/ca/uhn/fhir/util/MethodOutcomeUtil.java:12]: Nitpick: This class should have a private constructor

File: hapi-fhir-base/src/main/java/ca/uhn/fhir/util/MethodOutcomeUtil.java [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+
+	@Nonnull
+	public static Optional<String> getFirstResponseHeader(MethodOutcome theMethodOutcome, String theHeader) {
```
Comment (jamesagnew) [inline hapi-fhir-base/src/main/java/ca/uhn/fhir/util/MethodOutcomeUtil.java:15]: Actually, thinking even more, why not just put this method on MethodOutcome as `getFirstResponseHeader(String)`? That might be more discoverable.

**Review by jamesagnew — APPROVED** [review]

File: hapi-fhir-base/src/main/java/ca/uhn/fhir/util/MethodOutcomeUtil.java (hunk 2 above)
Comment (michaelabuckley) [reply hapi-fhir-base/src/main/java/ca/uhn/fhir/util/MethodOutcomeUtil.java:15]: Good suggestion.  No reason besides cowardice.  This sort of looked like a core lib, and I don't touch those.  I'll do that, and defend getResponseHeaders() against null.

File: hapi-fhir-base/src/main/java/ca/uhn/fhir/util/MethodOutcomeUtil.java (hunk 1 above)
Comment (michaelabuckley) [reply hapi-fhir-base/src/main/java/ca/uhn/fhir/util/MethodOutcomeUtil.java]: It's dead, Jim.

**Pull request merged**